main/src/main/scala/sbt/internal/DefaultBackgroundJobService.scala

```diff
@@ -306,6 +305,28 @@ private[sbt] abstract class AbstractBackgroundJobService extends BackgroundJobSe
       converter: FileConverter,
   ): Classpath =
     copyClasspath(products, full, workingDirectory, hashFileContents = true, converter)
+
+  private[sbt] def pauseChannelDuringJob(state: State, handle: JobHandle): Unit =
+    currentChannel(state) match
+      case Some(channel) =>
+        val level = channel.logLevel
+        channel.setLevel(Level.Error)
+        channel.pause()
+        handle match
+          case t: ThreadJobHandle =>
+            t.job.onStop: () =>
+              channel.setLevel(level)
+              channel.resume()
+              channel.prompt(ConsolePromptEvent(state))
+          case _ => ()
```

Comment:
Can the handle not be a `ThreadJobHandle`. If so the channel is never going to be resumed:

```
```
```suggestion
        handle match
          case t: ThreadJobHandle =>
            val level = channel.logLevel
            channel.setLevel(Level.Error)
            channel.pause()
            t.job.onStop: () =>
              channel.setLevel(level)
              channel.resume()
              channel.prompt(ConsolePromptEvent(state))
          case _ => ()
```

Reply:
I had this exact same concern. I think pausing in the match makes sense, and will prevent stuck terminals.